File: apps/frontend/lib/api/hunting.ts
import { apiFetch } from "./client";
import type { Session } from "./sessions";
import type { Finding } from "./findings";

export interface HuntingSessionCreate {
  project_id: string;
  preset_id: string;
  agent_id: string;
  commit_sha?: string | null;
  priority?: string;
  config?: Record<string, unknown>;
}

export interface PhaseUpdate {
  phase: string;
  status: string;
  data?: Record<string, unknown>;
}

export function createTargetDiscovery(
  data: HuntingSessionCreate,
): Promise<Session> {
  return apiFetch<Session>("/hunting/target-discovery", {
    method: "POST",
    body: JSON.stringify(data),
  });
}

export function createZeroDayHunt(
  data: HuntingSessionCreate,
): Promise<Session> {
  return apiFetch<Session>("/hunting/zero-day", {
    method: "POST",
    body: JSON.stringify(data),
  });
}

export function updatePhase(
  sessionId: string,
  data: PhaseUpdate,
): Promise<Session> {
  return apiFetch<Session>(`/hunting/sessions/${sessionId}/phase`, {
    method: "PATCH",
    body: JSON.stringify(data),
  });
}

export function fetchTargetCandidates(sessionId: string): Promise<Finding[]> {
  return apiFetch<Finding[]>(`/hunting/sessions/${sessionId}/targets`);
}

File: apps/backend/app/schemas/hunting.py
import uuid
from typing import Any

from pydantic import BaseModel

from app.models.analysis_session import SessionPriority


class HuntingSessionCreate(BaseModel):
    project_id: uuid.UUID
    preset_id: uuid.UUID
    agent_id: uuid.UUID
    commit_sha: str | None = None
    priority: SessionPriority = SessionPriority.NORMAL
    config: dict[str, Any] = {}


class PhaseUpdate(BaseModel):
    phase: str
    status: str
    data: dict[str, Any] | None = None
